src/components/UserForm.tsx

```diff
@@ -1,21 +1,29 @@
-import React from "react";
 import styled from "styled-components";
-
 import BaseButton from "./BaseButton";
 import BaseInput from "./BaseInput";
-import BaseIcon from "./BaseIcon";
-
-import eyeIcon from "../assets/icon/eye.svg";
-import eyeOffIcon from "../assets/icon/eye-off.svg";
-
-const UserFormGroup = ({ label, placeholder, className }) => {
-  const labelTable = {
-    email: "이메일",
-    username: "닉네임",
-    password: "비밀번호",
-    passwordCheck: "비밀번호 확인",
-  };
+import EyeIcon from "../assets/icon/eye.svg?react";
+import EyeOffIcon from "../assets/icon/eye-off.svg?react";
+
+interface UserFormGroupProps {
+  label: "email" | "username" | "password" | "passwordCheck";
+  placeholder: string;
+  errorMsg?: string;
+  className?: string;
+}
+
+const labelTable = {
```

Comment:
Label 타입을 따로 만드는건 어떨까요?

```suggestion
type Label = "email" | "username" | "password" | "passwordCheck";

const labelTable: Record<Label, string> = {
```
객체에 대한 타입 정의는 항상 신경써주는게 좋습니다 :)